### PR TITLE
WIP: fix CI memory issues

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- narrow scope of fixtures which load a model into memory to release the memory if it's no longer needed

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -510,24 +510,6 @@ async def trained_response_selector_bot(trained_async: Callable) -> Path:
     return Path(zipped_model)
 
 
-@pytest.fixture(scope="session")
-async def trained_restaurantbot(trained_async: Callable) -> Path:
```
Comment (wochinge) [inline tests/conftest.py:514]: moved this since it was only used by one module

**Comment by wochinge** [comment]
Seems this is giving us enough free memory. Currently also checking what stats we have on main (https://github.com/RasaHQ/rasa/pull/9554)
Screen Shot 2021-09-06 at 17 53 49

**Comment by wochinge** [comment]
That's the comparison on `main`
Screen Shot 2021-09-06 at 18 14 51

**Review by usc-m — APPROVED** [review]
LGTM

**Pull request merged**